zeebe/snapshot/src/main/java/io/camunda/zeebe/snapshots/impl/FileBasedSnapshotStoreImpl.java

```diff
@@ -358,16 +359,21 @@ public Either<SnapshotException, TransientSnapshot> newTransientSnapshot(
         new FileBasedSnapshotId(index, term, processedPosition, exportedPosition, brokerId);
 
     final FileBasedSnapshot currentSnapshot = this.currentSnapshot.get();
-    if (currentSnapshot != null && currentSnapshot.getSnapshotId().compareTo(newSnapshotId) == 0) {
+    if (currentSnapshot != null && currentSnapshot.getSnapshotId().equals(newSnapshotId)) {
       final String error =
           String.format(
               "Previous snapshot was taken for the same processed position %d and exported position %d.",
               processedPosition, exportedPosition);
       return Either.left(new SnapshotAlreadyExistsException(error));
     }
-    // transient snapshots are directly written to our snapshot dir
-    // with the sfv checksum file they are marked as valid
-    final var directory = buildSnapshotDirectory(newSnapshotId, false);
+    // transient snapshots are first written to a temporary directory and then later moved to the
```

Comment:
:slightly_smiling_face:  Dejavu! It used to be like this. I don't remember the reason for changing it. May be we should look it up just to ensure we are not re-introducing any issues.

Reply:
Haha yeah I remember that as well, asked about it here: https://camunda.slack.com/archives/C037RS2JHB8/p1752743278048189
From that it seems like the problem was maybe that we took transient snapshots in `/tmp/` which meant we couldn't do an atomic move at the end.